### PR TITLE
Add transform to dynamically create label layers

### PR DESCRIPTION
**Pull request by rmarianski**
Refs https://github.com/mapzen/tilequeue/issues/21

### CODE REVIEW

**Comment by rmarianski** [comment]
@zerebubuth Could you review please?

**Comment by zerebubuth** [comment]
:+1: looks good to me

**Comment by rmarianski** [comment]
Changes made to include the label features in the same source layer.

@zerebubuth, @nvkelso Look ok now?

File: TileStache/Goodies/VecTiles/transform.py [hunk 1]
```diff
@@ -1129,3 +1129,33 @@ def exterior_boundaries(feature_layers, zoom,
...
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        # shapely does the right thing for all kinds of geometries
+        # it also has a function `representative_point` which we might
+        # want to consider using too
+        label_centroid = shape.centroid
```
Comment (nvkelso) [inline TileStache/Goodies/VecTiles/transform.py:1150]: mapshaper is better at this, btw. We've been using it for Who's On First and it more reliably results in a lat, lng pair in the "middle" of a feature. But that introduces a new dependency, and is an optimization – so let's add a ticket to the backlog for this.

**Comment by zerebubuth** [comment]
Yup, :+1: looks good to me.

**Pull request merged**